File: C#/백준/Silver/15577. Prosjek/Prosjek.cs
using System;

namespace q{
    class w{
        static void Main(){
            int a = int.Parse(Console.ReadLine());
            float[] aa = new float[a];
            for (int i=0;i<a;i++){
                aa[i] = float.Parse(Console.ReadLine());
            }
            Array.Sort(aa);
            float avg = 0;
            for (int i=0;i<a-1;i++){
                avg = (aa[i]+aa[i+1])/2;
                aa[i+1]=avg;
            }
            Console.WriteLine(avg);
        }
    }
}

File: Java/백준/Silver/15577. Prosjek/Prosjek.java
import java.util.Scanner;
import java.util.Arrays;

public class Main{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int a = in.nextInt();
        float[] b = new float[a];
        for (int i=0;i<a;i++){
            b[i] = in.nextFloat();
        }
        Arrays.sort(b);
        float avg = 0;
        for (int i=0;i<a-1;i++){
            avg = (b[i]+b[i+1])/2;
            b[i+1] = avg;
        }
        System.out.println(avg);
    }
}